### DISCUSSION TITLE
[uno-check] detects Net SDK 7.0.402 but tries to install 7.0.401

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#14028] [General]
I am setting up a new computer.

I've installed VS 17.8.0 Preview 4.0 and then I installed UNO components.

When I run uno-check it detects Net SDK 7.0.402 but tries to install 7.0.401

```
> .NET SDK Checkup...
  - 7.0.402 - C:\Program Files\dotnet\sdk\7.0.402
  - 8.0.100-rc.2.23502.2 - C:\Program Files\dotnet\sdk\8.0.100-rc.2.23502.2

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
¤ Recommendation: Download .NET SDK (7.0.401)
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

! Attempt to fix? [y/n] (y): y
» Attempting to fix: .NET SDK
  Downloading .NET SDK 7.0.401...
  Downloading... 10%...
  Downloading... 20%...
  ...
  Downloading... 90%...
  Downloading... 100%...
  Downloaded .NET SDK 7.0.401.
  Installing .NET SDK 7.0.401...
  Installation skipped as a more recent version is installed .NET SDK 7.0.401. (See log file for more details:
C:\Users\mcnets\AppData\Local\Temp\tmpFE85.tmp)
Fix applied.  Checking again...
```

And always tries to install the new UNO templates that are already installed.

Then it skips net sdk workloads due the previous issue with the net sdk version.

```
× Skipped: .NET SDK - Workloads (7.0.401)
```

That's what dotnet --list-sdks returns:

```
➜ dotnet --list-sdks
7.0.402 [C:\Program Files\dotnet\sdk]
8.0.100-rc.2.23502.2 [C:\Program Files\dotnet\sdk]
```

**Comment by jeromelaban** [comment]
This is expected, uno.check uses explicit .NET versions and skips if a newer one is available. 7.0.402 is available in uno.check 1.16.1 that is being published on nuget as of this writing.

**Reply by mcNets** [reply]
That's why I can't find '7.0.401' in the actual uno-check repo 👍

**Comment by paulnsk** [comment]
The latest version is now 7.0.403 but uno-check is looking for 402. It does download it but fails to install, which, as @jeromelaban said, is the expected behavior. However, it then skips other checks:

![image](https://github.com/unoplatform/uno/assets/39160578/3271b947-6c0c-4756-ad5c-d7c8f1aba878)


As a result, my library no longer compiles:

![image](https://github.com/unoplatform/uno/assets/39160578/90720e7e-3c5a-45a6-abc1-c8acdf69500e)

By the way, this happens quite regularly: workloads 'disappear,' and uno-check fixes it. I've been okay with running uno-check every time this happens until this time when it can't fix the issue. Now, I wonder why the platform-specific workloads keep disappearing in the first place.

Please advise.  Can this be re-opened or should I create a new one?
Thanks.

**Reply by jeromelaban** [reply]
At this time, workloads with .NET 7 are behaving erratically, and .NET 8 improves things significantly. In particular, `dotnet workload clean --all` can help reset the whole workload situation for all versions of the SDKs, helping reinstall everything properly.

.NET 8 will be released very soon, unless you're willing to try with RC2 and VS 17.8 Preview 6.
